### PR TITLE
feat(nxls): decouple daemon lifecycle from nxls lifecycle

### PR DESCRIPTION
**Pull request by MaxKless**
- **log enabled**
- **changes**
- **start daemon separately & dont kill on shutdown**

### CODE REVIEW

File: apps/nxls-e2e/src/nxls-wrapper.ts [hunk 1]
```diff
@@ -134,6 +137,14 @@ export class NxlsWrapper {
 
     this.process?.removeListener('exit', this.earlyExitListener);
 
+    try {
+      execSync(`npx nx@${version ?? defaultVersion} daemon --stop`, {
```
Comment (jaysoo) [inline apps/nxls-e2e/src/nxls-wrapper.ts:141]: Should there be a windowsHide here as well?

File: libs/vscode/lsp-client/src/lib/nxls-client.ts [hunk 2]
```diff
@@ -102,7 +102,7 @@ export class NxlsClient {
   }
 
   public async refreshWorkspace() {
-    window.withProgress(
+    await window.withProgress(
```
Comment (jaysoo) [inline libs/vscode/lsp-client/src/lib/nxls-client.ts:105]: Does this fix the spamming refresh issue?

**Review by jaysoo — APPROVED** [review]

File: apps/nxls-e2e/src/nxls-wrapper.ts (hunk 1 above)
Comment (MaxKless) [reply apps/nxls-e2e/src/nxls-wrapper.ts:141]: sure, not important tho as it's just the e2e

File: libs/vscode/lsp-client/src/lib/nxls-client.ts (hunk 2 above)
Comment (MaxKless) [reply libs/vscode/lsp-client/src/lib/nxls-client.ts:105]: yes!

**Pull request merged**